File: generated-code/python/codecraft/FileReadWrite/model/attack_properties.py
from stream_wrapper import StreamWrapper

class AttackProperties:
    """Entity's attack properties"""

    __slots__ = ("attack_range","damage","collect_resource",)

    attack_range: int
    damage: int
    collect_resource: bool

    def __init__(self, attack_range: int, damage: int, collect_resource: bool):
        self.attack_range = attack_range
        """Maximum attack range"""
        self.damage = damage
        """Damage dealt in one tick"""
        self.collect_resource = collect_resource
        """If true, dealing damage will collect resource from target"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "AttackProperties":
        """Read AttackProperties from input stream
        """
        attack_range = stream.read_int()
        damage = stream.read_int()
        collect_resource = stream.read_bool()
        return AttackProperties(attack_range, damage, collect_resource)
    
    def write_to(self, stream: StreamWrapper):
        """Write AttackProperties to output stream
        """
        stream.write_int(self.attack_range)
        stream.write_int(self.damage)
        stream.write_bool(self.collect_resource)
    
    def __repr__(self):
        return "AttackProperties(" + \
            repr(self.attack_range) + \
            ", " + \
            repr(self.damage) + \
            ", " + \
            repr(self.collect_resource) + \
            ")"

File: generated-code/ruby/codecraft/FileReadWrite/model/attack_properties.rb
module Model

# Entity's attack properties
class AttackProperties
    # Maximum attack range
    attr_accessor :attack_range
    # Damage dealt in one tick
    attr_accessor :damage
    # If true, dealing damage will collect resource from target
    attr_accessor :collect_resource

    def initialize(attack_range, damage, collect_resource)
        @attack_range = attack_range
        @damage = damage
        @collect_resource = collect_resource
    end

    # Read AttackProperties from input stream
    def self.read_from(stream)
        attack_range = stream.read_int()
        damage = stream.read_int()
        collect_resource = stream.read_bool()
        AttackProperties.new(attack_range, damage, collect_resource)
    end

    # Write AttackProperties to output stream
    def write_to(stream)
        stream.write_int(@attack_range)
        stream.write_int(@damage)
        stream.write_bool(@collect_resource)
    end

    def to_s
        string_result = "AttackProperties { "
        string_result += "attack_range: "
        string_result += @attack_range.to_s
        string_result += ", "
        string_result += "damage: "
        string_result += @damage.to_s
        string_result += ", "
        string_result += "collect_resource: "
        string_result += @collect_resource.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end